messaging/src/main/protos/Model.proto

```diff
@@ -67,6 +67,27 @@ message AttachmentWithThumbnail {
   Attachment thumbnail = 2;
 }
 
+// Introduction is an introduction to a potential direct Contact
+message Introduction {
+  bytes id = 1; // the id of the person to whom you're being introduced
+  string displayName = 2; // the display name of person to whom you're being introduced
+}
+
+// Details of an Introduction (attached to both a StoredMessage as well as a StoredIntroduction
+message IntroductionDetails {
+  enum IntroductionStatus {
+    PENDING = 0; // introduction has been made but has not yet been accepted/rejected
+    ACCEPTED = 1; // introduction has been accepted
+  }
+
+  ContactId to = 1; // the id of the contact to which we're being introduced
+  string displayName = 2; // the display name of the contact to whom we're being introduced. This
+                          // may change when accepting an Introduction to the same contact made by
+                          // someone else.
+  string originalDisplayName = 3; // the original display name of the introduction, this will never change
```

Comment:
@Kallirroi Per discussion with @Derekf5, once somebody accepts an introduction, if we had other introductions to that same contact but made by other people, the displayName on those introductions is updated to reflect the displayName from the accepted introduction. I added this field to keep a record of the displayName as it was originally introduced, so if we ever decide that we want to use that in the UI we can.